src/Symfony/Component/JsonStreamer/Tests/Fixtures/stream_writer/null.php

```diff
@@ -2,7 +2,7 @@
 
 return static function (mixed $data, \Psr\Container\ContainerInterface $valueTransformers, array $options): \Traversable {
     try {
-        yield 'null';
+        yield "null";
```

Comment:
Why did you do this change here and in other places?

Reply:
Yielding a string with `"` instead of `'` allows me to inject variables directly in strings, such as `yield "{$prefix1}\"name\":";`. 

Therefore, I'm yielding every string with `"` to be able to merge them easily (see `yieldString` and `flushYieldBuffer`).

Reply:
do we have test case where the name contains $foo to ensure those are properly encoded?

Reply:
Nice catch @nicolas-grekas! It wasn't covered, and string interpolation could have occurred. I now escape every `$` that are not explicitly needed (I added a `DummyWithDollarProperties` test case to make sure of that).